Dalamud/Game/Gui/Nameplates/NameplateGui.cs

```diff
@@ -0,0 +1,178 @@
+
+        // Return the object for its object id
+        var objectId = namePlateInfo.ObjectID.ObjectID;
+        return this.objectTable.SearchById(objectId) as T;
+    }
+
+    private IntPtr HandleSetPlayerNameplateDetour(IntPtr playerNameplateObjectPtr, bool isTitleAboveName, bool isTitleVisible, IntPtr titlePtr, IntPtr namePtr, IntPtr freeCompanyPtr, IntPtr prefixPtr, int iconId)
+    {
+
+        if (this.OnNameplateUpdate != null)
+        {
+            // Create NamePlateObject if possible
+            var namePlateObj = new NameplateObject(playerNameplateObjectPtr);
+
+            // Create new event
+            var nameplateInfo = new NameplateInfo
+            {
+                Title = MemoryHelper.ReadSeStringNullTerminated(titlePtr),
+                Name = MemoryHelper.ReadSeStringNullTerminated(namePtr),
+                FreeCompany = MemoryHelper.ReadSeStringNullTerminated(freeCompanyPtr),
+                Prefix = MemoryHelper.ReadSeStringNullTerminated(prefixPtr),
+                IsTitleAboveName = isTitleAboveName,
+                IsTitleVisible = isTitleVisible,
+                IconID = (StatusIcons)iconId,
+            };
+
+            // Invoke event
+            var eventArgs = new NameplateUpdateEventArgs(nameplateInfo, namePlateObj);
+            this.OnNameplateUpdate.Invoke(eventArgs);
+
+            if (eventArgs.HasChanged)
+            {
+                // Get new states
+                isTitleAboveName = nameplateInfo.IsTitleAboveName;
+                isTitleVisible = nameplateInfo.IsTitleVisible;
+                iconId = (int)nameplateInfo.IconID;
+
+                // Get new Title string content
+                var titleRaw = nameplateInfo.Title.Encode();
+                var titleNewRaw = nameplateInfo.Title.Encode();
+                if (!titleRaw.SequenceEqual(titleNewRaw))
+                    MemoryHelper.WriteSeString(titlePtr, nameplateInfo.Title);
+
+                // Get new Name string content
+                var nameRaw = nameplateInfo.Name.Encode();
+                var nameNewRaw = nameplateInfo.Name.Encode();
+                if (!nameRaw.SequenceEqual(nameNewRaw))
+                    MemoryHelper.WriteSeString(namePtr, nameplateInfo.Name);
+
+                // Get new Free Company string content
+                var freeCompanyRaw = nameplateInfo.FreeCompany.Encode();
+                var freeCompanyNewRaw = nameplateInfo.FreeCompany.Encode();
+                if (!freeCompanyRaw.SequenceEqual(freeCompanyNewRaw))
+                    MemoryHelper.WriteSeString(freeCompanyPtr, nameplateInfo.FreeCompany);
+
+                // Get new Prefix string content
+                var prefixRaw = nameplateInfo.Prefix.Encode();
+                var prefixNewRaw = nameplateInfo.Prefix.Encode();
+                if (!prefixRaw.SequenceEqual(prefixNewRaw))
+                    MemoryHelper.WriteSeString(prefixPtr, nameplateInfo.Prefix);
```

Comment:
Please do not write to the pointers.
The data comes from StringArrayData and you don't know how much space the game has allocated for these strings.
Instead, you should use `Unsafe.AsPointer` to get pointers of your encoded byte arrays and pass them to the Original function. There is no need to free them, since the data is on the stack.
The only thing you probably have to do is to make sure they have a null terminator at the end, so concat the arrays with `new byte[] { 0 }`.